phac-nml/irida-next
# Member expiration backend

## What does this PR do and why?
Fixes #230.
- Added a new `expires_at` column in the members table.
- Created a new scope `not_expired` for members.
- Updated policies to only retrieve non-expired members.

## Screenshots or screen recordings
N/A

## How to set up and validate locally
There are a couple ways to test:
1. Call the new scope via console
`Member.for_namespace_and_ancestors(@group).not_expired` or `Member.for_namespace_and_ancestors(@project.namespace).not_expired` and verify the members.
2. Even though this is not the fronted PR, you can log in as the user with an expired membership to make sure groups and projects are not displayed. Some areas to check are:
    - the groups/projects dashboards
    - transferring groups/projects drop-down list

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
